src/box/box.cc

```diff
@@ -3141,6 +3141,19 @@ box_session_push(const char *data, const char *data_end)
 	return rc;
 }
 
+API_EXPORT int
+box_iproto_send(uint64_t sid,
```

Comment:
Why is that in the public C API, if we don't even have sessions there anyway? I mean, there is no way to get `sid` in C. So this function is just unusable for C modules unless they will do hacks and pass `sid` via Lua C API somehow.

I would drop this from the public C API.

Reply:
I agree with you: I am currently writing a patch exporting `box.session.id` to C API — perhaps we could hold this PR until we merge it? 

But, FWIW, I don't think it will help, since right now even in Lua I have to use `box.session.on_connect` trigger to retrieve an IPROTO session identifier.

Do you suggest exporting this function in the patch introducing IPROTO handlers overriding (gh-7716)?

@locker RFC

Reply:
Why do you have to use `on_connect`? There is `id()` method:
```
tarantool> box.session.id()
---
- 1
...
```
I don't suggest anything. I only note that without IDs this new API is practically unusable. I can still vote it up if the patch will be technically fine.

Reply:
> Why do you have to use on_connect? There is id() method:

How do I get the ID of the session for a socket connection using `box.session.id`?

Reply:
> Why is that in the public C API, if we don't even have sessions there anyway? I mean, there is no way to get `sid` in C. So this function is just unusable for C modules unless they will do hacks and pass `sid` via Lua C API somehow.

We're planning to add `box_session_id()` to the C API a bit later, but still in 2.11. This is mentioned in the [design document](https://www.notion.so/tarantool/box-iproto-override-44935a6ac7e04fb5a2c81ca713ed1bce). If we can add and test it before implementing `box_session_id()`, I think it's fine.

@CuriousGeorgiy in the future, please make sure that all reviewers read and approve the design document to avoid misunderstanding.

Reply:
@sergepetrenko @Gerold103 I bring my apologies: it didn't come to my mind to add you to design document review process — I just thought it would be nice if you reviewed this patch, since you work in an adjacent area.

@sergepetrenko discussed with @locker: it seems significant to be able to send packets to arbitrary sessions, since the usages is not limited to broadcasts, but can only be some sort of subscription or multicast.